raiden/blockchain_events_handler.py

```diff
@@ -588,16 +588,32 @@ def handle_secret_revealed(raiden: "RaidenService", event: Event):
     block_number = data["block_number"]
     block_hash = data["block_hash"]
     transaction_hash = data["transaction_hash"]
-    registeredsecret_state_change = ContractReceiveSecretReveal(
-        transaction_hash=transaction_hash,
-        secret_registry_address=secret_registry_address,
-        secrethash=args["secrethash"],
-        secret=args["secret"],
-        block_number=block_number,
-        block_hash=block_hash,
-    )
-
-    raiden.handle_and_track_state_change(registeredsecret_state_change)
+    secret=args["secret"]
+    secrethash=args["secrethash"]
+    chain_state = views.state_from_raiden(raiden)
+    if chain_state.get_payment_task(raiden.address, secrethash):
+        registeredsecret_state_change = ContractReceiveSecretReveal(
+            transaction_hash=transaction_hash,
+            secret_registry_address=secret_registry_address,
+            secrethash=secrethash,
+            secret=secret,
+            block_number=block_number,
+            block_hash=block_hash,
+        )
+        raiden.handle_and_track_state_change(registeredsecret_state_change)
+    else:
+        for lc_address, payment_state in chain_state.get_payment_states_by_address():
+            if secrethash in set(payment_state.secrethashes_to_task):
+                registeredsecret_state_change = ContractReceiveSecretRevealLight(
+                    transaction_hash=transaction_hash,
+                    secret_registry_address=secret_registry_address,
+                    secrethash=secrethash,
+                    secret=secret,
+                    block_number=block_number,
+                    block_hash=block_hash,
+                    lc_address=lc_address
+                )
+                raiden.handle_and_track_state_change(registeredsecret_state_change)
```

Comment:
Here we are checking if the node is a hub, and sending the event to the light clients if so. I don't really like the _for_, but can't think right now  a better way

Reply:
why not accessing to chain_state.get_payment_states_by_address()[lc_address] here?

youre iterating over all the payment tasks right? even the tasks that are not related with the lc

Reply:
The problem with that is we don't know which lc is related to the payment task. Our only input is the payment's secrethash

Reply:
got it, lets keep it like this then

what do you guys think? @mortelli @JONAF2103

Reply:
I don't know about this but ok if you approve it @marcosmartinez7 then i think is ok, but why is only the secret hash available here? i mean if we have that we should have the address too